### PR TITLE
Add micro-blog exercise

### PR DESCRIPTION
**Pull request by ccouzens**
This is an exercise requiring students to truncate unicode strings. #1507

### CODE REVIEW

**Comment by sshine** [comment]
While the description of UTF-8 and UTF-16 are very helpful, will they cause confusion for tracks that go with either one where the student is not fully aware which is being used? If a track somehow makes it explicit which encoding is chosen, would this be recommendable to place in a track-specific hint?

**Comment by ccouzens** [comment]
Hi @sshine ,

Thanks for the feedback :)

I agree that it makes sense to have the language mention if they represent UTF-8 or UTF-16.

Are you thinking something like the following addition to `canonical-data -> comments`?

> Consider adding a track specific hint.md about how your language handles strings internally.

**Comment by cmccandless** [comment]
Love the name and concept!

**Comment by ccouzens** [comment]
@sshine I've pushed a new commit with a bit about track specific hints to say what applies to the current language.

https://github.com/exercism/problem-specifications/commit/251246edd87429fb525026763fa94df3608a8035#diff-ea7d2610e08991cff203e0a81d289ae4R18

**Comment by ccouzens** [comment]
Hello,

Is there anything I should be doing, or should I just be patient? 🙂

**Comment by SaschaMann** [comment]
I think it would be nice to add some test cases that aren't emoji or English - perhaps cases with germanic umlauts, cyrillic and/or greek letters, historic scripts etc. - because that's one of the main uses and goals of unicode.

**Comment by SleeplessByte** [comment]
I agree with that. But perhaps no graphèmes/compound characters as those are a whole thing by themselves.

File: exercises/micro-blog/description.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+characters are precisely 1 byte long.
+* **UTF-8** is a Unicode text encoding. Characters take between 1
+and 4 bytes.
+* **UTF-16** is a Unicode text encoding. Characters are either 2 or
+4 bytes long.
+
+UTF-8 and UTF-16 are both Unicode encodings which means they're capable of
+representing a massive range of characters including:
+* English language text
+* Foreign language text
```
Comment (SaschaMann) [inline exercises/micro-blog/description.md]: I don't like that this assumes the perspective of a native English speaker. English is a foreign language to most of the world. Perhaps something along the lines of "text in most of the world's languages and scripts" would be a better description.

Comment (SleeplessByte) [reply exercises/micro-blog/description.md]: But UTF-8 was designed to be best with English so that ís something that can be mentioned.

Comment (SaschaMann) [reply exercises/micro-blog/description.md]: It definitely can be mentioned that UTF-8 was designed to be compatible to ASCII but that can be phrased in a better way that doesn't have the issue above.

**Comment by ccouzens** [comment]
I initially tried some Egyptian Hieroglyphics but found they didn't display correctly in my terminal or text editor. I'll try some of the other suggestions (germanic umlauts, cyrillic and/or greek letters)

File: exercises/micro-blog/description.md (hunk 1 above)
Comment (ccouzens) [reply exercises/micro-blog/description.md]: I've incorporated the suggestion about not assuming native English speakers d2f65b4c29ea1710f9d8fe0e0dcb44fc948a3b90

I've not put in anything to do with UTF-8 being able to represent English in the least space or its backwards compatibility with ASCII. Whilst they're nice things to know about UTF-8, I'm not sure they'd help students solve the exercise.

**Comment by ccouzens** [comment]
I've added test cases that aren't English or Emoji 786c1e69cf7b4af8ae205a165290a387a56af6dd

I'd like to have examples in other languages that do something clever or amusing when truncated to 5 characters. Unfortunately, I don't think I'll be able to come up with any myself, but if someone suggests some I'll add them.

**Comment by SaschaMann** [comment]
> I'd like to have examples in other languages that do something clever or amusing when truncated to 5 characters. Unfortunately, I don't think I'll be able to come up with any myself, but if someone suggests some I'll add them.

I've asked around on an IRC server and someone came up with "Bärteppich" which is German and describes a "bear carpet", as in a carpet made of bear fur. Truncated to 5 letters it's "Bärte", which means "beards". Just make sure to use the single-character `ä` and not the compound `ä` when adding it.

**Comment by rpottsoh** [comment]
@ccouzens 
> Is there anything I should be doing

[New exercises require a glyph](https://github.com/exercism/problem-specifications#new-exercises-require-a-glyph).

**Comment by ccouzens** [comment]
@SaschaMann thank you for that suggestion- it's brilliant and I've added it in
ee7fcce7695bdf6056d0cefd295f8956c8b2c9a2

**Comment by ccouzens** [comment]
Thank you @rpottsoh - I've created an issue in the website-icons github project https://github.com/exercism/website-icons/issues/29

**Comment by SleeplessByte** [comment]
I'm very excited about this. 

I know this is out of scope for this exercise, but just so we can refer back to it, and/or use it as a starting point, perhaps there is an extension exercise where we can add:
- diacritics ([combining character](https://en.wikipedia.org/wiki/Combining_character))
- [ligatures](https://en.wikipedia.org/wiki/Typographic_ligature)
- compound characters, regional flag characters and/or unicode conversion

**Comment by SleeplessByte** [comment]
I think this is mostly waiting on https://github.com/exercism/website-icons/issues/29 :)

**Comment by ccouzens** [comment]
The icons are merged 😃

https://github.com/exercism/website-icons/blob/d9687ea6945bd6714ac6d2240f756a3bc5fda143/exercises/micro-blog-dark.png
https://github.com/exercism/website-icons/blob/d9687ea6945bd6714ac6d2240f756a3bc5fda143/exercises/micro-blog-turquoise.png
https://github.com/exercism/website-icons/blob/d9687ea6945bd6714ac6d2240f756a3bc5fda143/exercises/micro-blog-white.png

Is there something I should be doing such as rebasing, or is this now good to go?

**Comment by rpottsoh** [comment]
Based on the age of this PR and has the requisite approvals, this PR should probably be merged.  Do you concur @ihid?

**Comment by iHiD** [comment]
I do concur. Thank you for checking, @rpottsoh.

**Comment by SleeplessByte** [comment]
🎉

**Pull request merged**